### PR TITLE
Add HAProxy Agent Check

### LINKED ISSUE #1196: Drain a backend in terminating status?
We have a legacy application ( tomcat/java), which needs sticky sessions. When we deploy new versions of our applications, we need to stop sending new connections to a server, while sending bound sessions to the old server. Please note: this is not referring to in-flight requests, we're needing the active tomcat sessions to expire, which normally takes a few hours.

This is possible using haproxy drain command. This will send bound connections to the old server, but send new ones elsewhere. 

When kubernetes terminates a pod, it enters the TERMINATING status. When sticky sessions are enabled, the desired functionality is usually to put the associated pod into drain mode.  This is nice because it allows the application preStop hooks to decide when its ready to stop. When the preStop hook terminates, the pod dies, and bound sessions end.

How could we accomplish this flow using voyager? We are currently using [jcmoraisjr/haproxy-ingress](https://github.com/jcmoraisjr/haproxy-ingress), because it meets this requirement. ( In fact, the developers there were kind enough to add this feature.  I'm sure [their implementation](https://github.com/jcmoraisjr/haproxy-ingress/pull/95) could be ported, presuming no licensing issues exist.

### PR DESCRIPTION
**Pull request by kfoozminus**
http://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.2-agent-check

- Add agent-check support
- Update docs
- Add sample agent-check server to appscode/test-server
- Push test-server image: appscode/test-server:2.4
- Add e2e test for agent-check
- Add apache bench to travis.yml

Fixes #1196 #1210

### CODE REVIEW

File: apis/voyager/v1beta1/annotations.go [hunk 1]
```diff
@@ -255,6 +255,11 @@ const (
 	// ref: https://github.com/appscode/voyager/issues/1054
 	NodeSelector = EngressKey + "/" + "node-selector"
 	Tolerations  = EngressKey + "/" + "tolerations"
+
+	// https://github.com/appscode/voyager/issues/1210
+	// http://cbonte.github.io/haproxy-dconv/1.8/configuration.html#agent-check
+	AgentPort  = EngressKey + "/" + "agent-port"
+	AgentInter = EngressKey + "/" + "agent-inter"
```
Comment (tamalsaha) [inline apis/voyager/v1beta1/annotations.go]: Use full words `agent-interval`

Comment (kfoozminus) [reply apis/voyager/v1beta1/annotations.go]: Done.

File: test/framework/ingress_util.go [hunk 2]
```diff
@@ -1309,3 +1320,58 @@ func (i *ingressInvocation) GetServiceWithLoadBalancerStatus(name, namespace str
...
+	var pods *core.PodList
+	Eventually(func() error {
+		pods, err = i.KubeClient.CoreV1().Pods(i.TestNamespace).List(metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(label).String(),
+		})
+		return err
+	}, "10m", "10s").Should(BeNil())
+
+	// select the first pod and get the node in which this pod is assigned
+	node, err := i.KubeClient.CoreV1().Nodes().Get(pods.Items[0].Spec.NodeName, metav1.GetOptions{})
```
Comment (tamalsaha) [inline test/framework/ingress_util.go]: List Nodes and use Node[0]

File: test/e2e/agent_check.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+			By("Requesting With Cookie Should Be OK")
+			err = f.Ingress.DoHTTPStatusWithCookies(5, ing, eps, "GET", "/testpath/ok", []*http.Cookie{lastCookie}, func(r *client.Response) bool {
+				return Expect(r.Status).Should(Equal(http.StatusOK))
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Describe("With Wrong Port and Default Agent Inter", func() {
```
Comment (tamalsaha) [inline test/e2e/agent_check.go]: Describe the test case - agent check missing and uses `check`

File: test/e2e/agent_check.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	})
+
+	Describe("With Correct Port and Default Agent Inter", func() {
+		BeforeEach(func() {
+			svcAnnotation = map[string]string{
+				api.AgentPort: "5555",
+			}
+		})
+
+		It("Should Response HTTP", func() {
```
Comment (tamalsaha) [inline test/e2e/agent_check.go]: combine the two tests.

File: test/e2e/agent_check.go [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+	Describe("With Correct Port and Customized Agent Inter", func() {
+		BeforeEach(func() {
+			svcAnnotation = map[string]string{
+				api.AgentPort:     "5555",
+				api.AgentInterval: "1s",
+			}
+		})
+
+		It("Should Response HTTP", func() {
```
Comment (tamalsaha) [inline test/e2e/agent_check.go]: one test

File: test/e2e/agent_check.go [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+				}
+			}
+
+			By("Requesting Without Cookie Should Not Respond")
+			err = f.Ingress.DoHTTPStatus(5, ing, eps, "GET", "/testpath/ok", func(r *client.Response) bool {
+				return Expect(r.Status).Should(Equal(http.StatusServiceUnavailable))
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Requesting With Cookie Should Be OK")
```
Comment (tamalsaha) [inline test/e2e/agent_check.go]: Explain why cookie should be ok.

**Pull request merged**